silhouette/src/main/scala/silhouette/http/decoder/Decoder.scala

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Minutemen Group under one or more contributor license
+ * agreements. See the COPYRIGHT file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package silhouette.http.decoder
+
+import scala.util.Try
+
+/**
+ * Represents a decode action that extract from A an instance of B
+ *
+ * @tparam A The raw data source of decode action.
+ * @tparam B The type target of decode action.
+ */
+trait Decoder[A, B] {
+
+  /**
+   * Decode from A to target B.
+   *
+   * @param in The raw data source of decode action.
+   * @return a [[scala.util.Success]] with an instance of B
+   *         or a [[scala.util.Failure]] with error
+   */
+  def decode(in: A): Try[B]
```

Comment:
I think it should be only:
```scala
def decode(in: A): B
```

The `Try` can be added also in the type param definition. I know that mostly a `Try` is needed because decode throws mostly an error. But it may also possible to use a `Future`.